### PR TITLE
feat: Support go native git as fallback for git binary

### PR DESCRIPTION
**Pull request by roneli**

### CODE REVIEW

File: internal/getmodules/get_git.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import (
+	"net/url"
+	"os/exec"
+
+	"github.com/hashicorp/go-getter"
+)
+
+type GitGetter struct {
+	getter.GitGetter
```
Comment (disq) [inline internal/getmodules/get_git.go:11]: Why embed both Getters in the struct? I would give them names so their methods aren't implicitly used/made available by something like `getmodules.GitGetter.SomeMethodInHashiGitGetter()`

```
type GitGetter struct {
	hashiImpl getter.GitGetter
	goImpl GoGitGetter
}
```
or something like that

**Review by disq — CHANGES_REQUESTED** [review]

File: internal/getmodules/go_git.go [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+		Name:  "",
+		URLs:  []string{u.String()},
+		Fetch: nil,
+	}).List(&git.ListOptions{
+		Auth:            nil,
+		InsecureSkipTLS: false,
+		CABundle:        nil,
+	})
+	if err != nil {
+		return "master"
```
[the remark this answers is not in the document]
Comment (roneli) [reply internal/getmodules/go_git.go:236]: I left it master as that is the original code, I can change it to main, was thinking that as well, wdyt?

[the remark this answers is not in the document]
Comment (disq) [reply internal/getmodules/go_git.go:236]: That's what the suggestion is all about :)

**Comment by yevgenypats** [comment]
Can we close this as we are not going to support it soon?

**Comment by erezrokah** [comment]
Closing as stale, if I got this wrong please let me know and I'll re-open

**Pull request closed without merging**